sidecar-ffi/src/lib.rs

```diff
@@ -642,3 +642,12 @@ pub unsafe extern "C" fn ddog_sidecar_dogstatsd_set(
 
     MaybeError::None
 }
+
+#[no_mangle]
+#[allow(clippy::missing_safety_doc)]
```

Comment:
I believe we should document the propose and arguments of all public ffi APIs.

Reply:
👍 
I added a few comments. I suggest to continue in separate PRs.

Reply:
Very nice. Thanks for adding comments to things you didn't change :)